ui/src/components/VoxettaRecordButton.js

```diff
@@ -0,0 +1,144 @@
+    static get properties() {
+        return {
+            isRecording: {type: Boolean},
+            audioStream: {type: Object}
+        };
+    }
+
+    static get styles() {
+        return css`
+            div {
+                align-items: center; 
+                display: flex; 
+                flex-direction: column; 
+                flex-wrap: wrap;
+                justify-content: center; 
+                text-align: center;  
+            }
+            mwc-icon-button {
+                --mdc-icon-button-size: 80px;
+                --mdc-icon-size: 40px; 
+                border-radius: 50%;
+                box-shadow: 0px 8px 15px #dcdcdc;
+                color: #3c4043;
+            }
+        `;
+    }
+
+    constructor() {
+        super();
+        this.isRecording = false;
+        this.audioRecorder = new AudioRecorder();
+        this.utteranceService = new UtteranceApiService();
+        this.audioStream; 
+    }
+
+    updated() {
+        this.handleWaveCanvas();
+    }
+  
+    /**
+     * If the user is not currently recording, begin recording using the Microphone 
+     * component. Otherwise, stop recording and save and display the just-recorded 
+     * audio file.
+     */
+    async recordHandler() {
+        if (!this.isRecording) {
+            try { 
+                await this.audioRecorder.initRecorder();
+            } catch(e) { 
+                console.log('Microphone blocked.'); 
+                return;
+            }
+
+            if (!this.audioRecorder.startRecording()) {
+                 console.log('Failed to start recording.')
+                 return; 
+            }
+            this.isRecording = true;
+            this.audioStream = this.audioRecorder.stream;
+        } else {
+            this.isRecording = false;
+            this.handleFinish();
+            const audio = await this.audioRecorder.stopRecording();
+            if (audio.recordingUrl) {
+                this.utteranceService.saveAudio(audio);
+            }
+        }
+    }
+
+    /**
+     * Emits an event that causes the application to render a sound
+     * wave that corresponds to the current audio stream. 
+     */
+    handleWaveCanvas() {
+        const event = new CustomEvent('update-wave', {
+            detail: { 
+                isRecording: this.isRecording,
+                audioStream: this.audioStream
+            }
+        });
+        this.dispatchEvent(event);
+    }
+
+    /**
+     * Emits an event that causes a new prompt to be rendered
+     * on the recording page. 
+     */
+    handleFinish() {
+        const event = new CustomEvent('change-prompt', {});
+        this.dispatchEvent(event);
+    }
+    
+    render() {
+        return html`
+            <mwc-icon-button 
+                id="record-button"
+                icon=${this.isRecording ? "stop" : "mic"}
+                style=${styleMap(this.isRecording ? recordingStyle : nonRecordingStyle)}
```

Comment:
would this be more performant if you just had this mapped to styles instead of changing the actual style property?